bootloader/bootloader.go

```diff
@@ -352,8 +352,17 @@ func ForceError(err error) {
 	forcedError = err
 }
 
-func extractKernelAssetsToBootDir(dstDir string, snapf snap.Container, assets []string) error {
-	// now do the kernel specific bits
+// extractKernelAssetsToBootDir will extract the given assets into the
+// given assetsDir. If the target directory already exists it will do
+// nothing.
+func extractKernelAssetsToBootDir(assetsDir string, snapf snap.Container, assets []string) error {
+	// Final dir already in place: nothing more to do.
+	if st, err := os.Stat(assetsDir); err == nil && st.IsDir() {
+		return nil
+	}
+
+	// unpack kernel assets into tmpdir
+	dstDir := assetsDir + ".unpacking"
```

Comment:
we should probably RemoveAll that dir in case before creating/writing to it?

Reply:
also we probably need recover code here etc to remove this dir if there's an error?

Reply:
We had remove on error originally, happy to add it back. But then the existing code is not doing this either and relies on the caller to cleanup. This is done via removeKernelAssetsFromBootDir() in error cases which will DTRT.

Also happy to add RemoveAll() before. We run unsquahfs with -f so it should be fine but better safe then sorry :)